r-package/grf/R/causal_survival_forest.R

```diff
@@ -0,0 +1,758 @@
+#' Causal survival forest (experimental)
+#'
+#' Trains a causal survival forest that can be used to estimate
+#' conditional average treatment effects theta(X). When the treatment assigment
+#' is unconfounded, we have theta(X) = E[T(1) - T(0) | X = x],
+#' where T is the survival time up to a fixed maximum follow-up time.
+#' T(1) and T(0) are potental outcomes corresponding to the two possible
+#' treatment states.
+#'
+#' @param X The covariates.
+#' @param Y The event time (may be negative).
+#' @param W The treatment assignment (must be a binary vector with no NAs).
+#' @param D The event type (0: censoring, 1: failure).
+#' @param W.hat Estimates of the treatment propensities E[W | Xi]. If W.hat = NULL,
+#'              these are estimated using a separate regression forest. Default is NULL.
+#' @param S1.hat Estimates of the conditional survival function S(t, x, 1) = P[Y > t | X = x, W = 1].
```

Comment:
Would we plausibly expect users to pass in values for `S1.hat`, `S0.hat`, `C.hat`, `lambda.C.hat`? If not, we could avoid making them configurable to reduce the API surface.

Reply:
It is for consistency with other forests that has nuisance parameters, like `Y.hat` and `W.hat` the user may want to fine tune (change min.node.size etc), or for example, a user can threshold low inverse probability weighting of censoring weights by doing
```
sf.censor = survival_forest(X, Y, 1 - D, prediction.type = "Nelson-Aalen")
C.hat = predict(sf.censor, failure.times = sort(unique(Y)))$predictions
C.hat[C.hat < 0.01] = 0.01
cs.forest = causal_survival_forest(X, Y, W, D,
                                   C.hat = C.hat)
```

and for ` lambda.C.hat`, a user may fine-tune this by calculating derivates from a spline fit instead of the default forward difference.

Reply:
Okay, this makes sense. I also caught up with @swager and he confirms that it's helpful to be able to pass these in (even just for performance reasons -- you can calculate nuisance parameters once, then try several different forests).